### ISSUE TITLE
QualifierAlignment Right doesn't handle Pointer to member declarators correctly

### ISSUE THREAD

**Issue by AlexanderHederstaf** [llvm/llvm-project#60898]
Example input:

```console
$ cat /tmp/memberpointer.cpp
struct S;
void foo(const int S::*const ptr) {}
void bar(int const S::*const ptr) {}
```

Output:
```console
$ ./clang-format  --style="{QualifierAlignment: Right}" /tmp/memberpointer.cpp
struct S;
void foo(int const S::*const ptr) {}
void bar(int S::const *const ptr) {}
```

Discovered by running Clang-Format on an existing codebase that had a few instances of pointer to member function. The `const` qualifier is moved iteratively until the code no longer compiles. Probably caused in this section

```cpp
    // The case  `const Foo` -> `Foo const`
    // The case  `const ::Foo` -> `::Foo const`
    // The case  `const Foo *` -> `Foo const *`
    // The case  `const Foo &` -> `Foo const &`
    // The case  `const Foo &&` -> `Foo const &&`
    // The case  `const std::Foo &&` -> `std::Foo const &&`
    // The case  `const std::Foo<T> &&` -> `std::Foo<T> const &&`
    while (Next && Next->isOneOf(tok::identifier, tok::coloncolon))
      Next = Next->Next;
```

Where special treatment would be required for the following case, causing no change.
```cpp
    // The case  `const Foo::*` -> `const Foo::*`
```

**Comment by rymiel** [comment]
https://reviews.llvm.org/D144537

**Comment by AlexanderHederstaf** [comment]
I tried to also solve https://github.com/llvm/llvm-project/issues/57154 as well by refactoring and taking a slightly different approach with testing the previous tokens. I uploaded a review here https://reviews.llvm.org/D144709. This patch succeeds in always moving qualifiers to the right but fails unit-tests related to the ordering of const/volatile. I also did not make any changes to west-const and did not yet add any new tests.

Worth noting is that the cases that fail now for `int` would fail on main for `Foo`.
